### ISSUE TITLE
NAMESERVER_IP_WITHOUT_REVERSE emitted even when PTR record exists

### ISSUE THREAD

**Issue by anandb-ripencc** [zonemaster/zonemaster-engine#1105]
```
$ zonemaster-cli 15.248.151.in-addr.arpa --raw
Loading profile from /etc/zonemaster/ripencc.json.
   1.77 WARNING   NAMESERVER_IP_WITHOUT_REVERSE   ns_ip=185.110.23.0; nsname=ns1.nbit.it
   2.80 WARNING   IPV4_ONE_ASN   asn=12874
   6.89 NOTICE    NO_IPV6_NS_CHILD   count=0; minimum=2; ns_ip_list=; nsname_list=
   6.89 NOTICE    NO_IPV6_NS_DEL   count=0; minimum=2; ns_ip_list=; nsname_list=
   6.97 NOTICE    NEITHER_DNSKEY_NOR_DS   child=185.110.20.9; parent=2001:500:14:6100:ad::1
   6.97 NOTICE    NOT_SIGNED   zone=15.248.151.in-addr.arpa
   8.96 NOTICE    NO_MX_RECORD

$ host 185.110.23.0
0.23.110.185.in-addr.arpa domain name pointer uo1.nbit.it.
```

**Comment by matsduf** [comment]
I assume that you run the latest version. I cannot replicate the issue:
```
$ zonemaster-cli 15.248.151.in-addr.arpa --raw
   2.70 NOTICE    ADDRESS03      NAMESERVER_IP_PTR_MISMATCH   names=s1.extnet.it.; ns_ip=185.110.20.9; nsname=ns1.extnet.it
   2.70 NOTICE    ADDRESS03      NAMESERVER_IP_PTR_MISMATCH   names=uo1.nbit.it.; ns_ip=185.110.23.0; nsname=ns1.nbit.it
   3.86 WARNING   CONNECTIVITY03 IPV4_ONE_ASN   asn=12874
   9.26 NOTICE    DELEGATION01   NO_IPV6_NS_CHILD   count=0; minimum=2; ns_ip_list=; nsname_list=
   9.26 NOTICE    DELEGATION01   NO_IPV6_NS_DEL   count=0; minimum=2; ns_ip_list=; nsname_list=
   9.35 NOTICE    DNSSEC07       NEITHER_DNSKEY_NOR_DS   child=185.110.20.9; parent=2001:500:14:6100:ad::1
   9.35 NOTICE    UNSPECIFIED    NOT_SIGNED   zone=15.248.151.in-addr.arpa
  11.43 NOTICE    ZONE09         NO_MX_RECORD   
```
```
$ zonemaster-cli --version
CLI version:    v4.0.1
Engine version: v4.5.1
(...)
```

**Comment by anandb-ripencc** [comment]
Yes, I'm using the latest version, and I still keep getting this warning.
```
$ zonemaster-cli --version
CLI version:    v4.0.1
Engine version: v4.5.1
...

$ zonemaster-cli 15.248.151.in-addr.arpa --raw
Loading profile from /etc/zonemaster/ripencc.json.
   1.93 WARNING   NAMESERVER_IP_WITHOUT_REVERSE   ns_ip=185.110.23.0; nsname=ns1.nbit.it
   2.93 WARNING   IPV4_ONE_ASN   asn=12874
   7.01 NOTICE    NO_IPV6_NS_CHILD   count=0; minimum=2; ns_ip_list=; nsname_list=
   7.01 NOTICE    NO_IPV6_NS_DEL   count=0; minimum=2; ns_ip_list=; nsname_list=
   7.08 NOTICE    NEITHER_DNSKEY_NOR_DS   child=185.110.20.9; parent=2001:500:14:6100:ad::1
   7.08 NOTICE    NOT_SIGNED   zone=15.248.151.in-addr.arpa
   9.08 NOTICE    NO_MX_RECORD
```

**Comment by matsduf** [comment]
Can you add `--show_testcase` to verify which test case that outputs the message?

**Comment by matsduf** [comment]
There are two test cases that can output the message `NAMESERVER_IP_WITHOUT_REVERSE`, Address02 and Address03, respectively. Below I include the output from the following two commands:
```
zonemaster-cli 15.248.151.in-addr.arpa --raw --show_testcase --test address/address02 --level DEBUG
zonemaster-cli 15.248.151.in-addr.arpa --raw --show_testcase --test address/address03 --level DEBUG
```
Can you run the same two commands?

```
$ zonemaster-cli 15.248.151.in-addr.arpa --raw --show_testcase --test address/address02 --level DEBUG
   0.00 DEBUG     UNSPECIFIED    START_TIME   string=2022-07-25 16:54:10 +0000; time_t=1658768050
   0.00 DEBUG     UNSPECIFIED    TEST_ARGS   args=Zonemaster::Engine::Zone=HASH(0x804fe5ba0); module=address; testcase=address02
   0.00 INFO      UNSPECIFIED    GLOBAL_VERSION   version=v4.5.1
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=Zonemaster::LDNS; version=2.2.2
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=IO::Socket::INET6; version=2.72
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=Moose; version=2.2201
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=Module::Find; version=0.15
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=File::ShareDir; version=1.118
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=File::Slurp; version=9999.27
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=Net::IP; version=1.26
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=List::MoreUtils; version=0.430
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=Clone; version=0.45
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=Readonly; version=2.05
   0.00 DEBUG     UNSPECIFIED    MODULE_VERSION   module=Zonemaster::Engine::Test::Address; version=v1.0.8
   0.00 DEBUG     ADDRESS02      TEST_CASE_START   testcase=address02
   0.01 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=15.248.151.in-addr.arpa; type=SOA
   0.04 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=15.248.151.in-addr.arpa; type=SOA
   0.04 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=rirns.arin.net; type=A
   0.08 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:501:b1f9::30; name=rirns.arin.net; type=A
   0.09 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=204.61.216.50; name=rirns.arin.net; type=A
   0.10 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=rirns.arin.net; type=AAAA
   0.13 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:501:b1f9::30; name=rirns.arin.net; type=AAAA
   0.14 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=204.61.216.50; name=rirns.arin.net; type=AAAA
   0.15 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.253.249.53; name=15.248.151.in-addr.arpa; type=SOA
   0.31 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=ns1.nbit.it; type=A
   0.34 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:1010:7::53; name=ns1.nbit.it; type=A
   0.35 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=5.196.13.82; name=ns1.nbit.it; type=A
   0.38 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=ns1.nbit.it; type=AAAA
   0.41 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:1010:7::53; name=ns1.nbit.it; type=AAAA
   0.42 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=5.196.13.82; name=ns1.nbit.it; type=AAAA
   0.45 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=185.110.23.0; name=15.248.151.in-addr.arpa; type=SOA
   0.52 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.253.249.53; name=248.151.in-addr.arpa; type=SOA
   0.67 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=151.in-addr.arpa; type=SOA
   0.71 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=151.in-addr.arpa; type=SOA
   0.72 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.253.249.53; name=151.in-addr.arpa; type=SOA
   0.88 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=in-addr.arpa; type=SOA
   0.91 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=in-addr.arpa; type=SOA
   0.91 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=arpa; type=SOA
   0.95 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=198.41.0.4; name=arpa; type=NS
   0.96 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=a.ns.arpa; type=A
   0.99 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=a.ns.arpa; type=AAAA
   1.03 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=198.41.0.4; name=in-addr.arpa; type=NS
   1.03 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=a.in-addr-servers.arpa; type=A
   1.07 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=a.in-addr-servers.arpa; type=A
   1.07 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=a.in-addr-servers.arpa; type=AAAA
   1.11 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=a.in-addr-servers.arpa; type=AAAA
   1.11 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.180.182.53; name=in-addr.arpa; type=NS
   1.27 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.180.182.53; name=151.in-addr.arpa; type=NS
   1.43 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=ns3.afrinic.net; type=A
   1.46 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:501:b1f9::30; name=ns3.afrinic.net; type=A
   1.47 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2620:38:2000::53; name=ns3.afrinic.net; type=A
   1.62 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=ns3.afrinic.net; type=AAAA
   1.66 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:501:b1f9::30; name=ns3.afrinic.net; type=AAAA
   1.66 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2620:38:2000::53; name=ns3.afrinic.net; type=AAAA
   1.82 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:500:14:6100:ad::1; name=151.in-addr.arpa; type=NS
   1.85 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:500:14:6100:ad::1; name=15.248.151.in-addr.arpa; type=NS
   1.86 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=ns1.extnet.it; type=A
   1.90 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:1010:7::53; name=ns1.extnet.it; type=A
   1.90 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=5.196.13.82; name=ns1.extnet.it; type=A
   1.94 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=ns1.extnet.it; type=AAAA
   1.97 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:1010:7::53; name=ns1.extnet.it; type=AAAA
   1.97 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=5.196.13.82; name=ns1.extnet.it; type=AAAA
   2.02 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=185.110.20.9; name=15.248.151.in-addr.arpa; type=NS
   2.09 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=9.20.110.185.in-addr.arpa; type=PTR
   2.12 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=9.20.110.185.in-addr.arpa; type=PTR
   2.14 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.253.249.53; name=9.20.110.185.in-addr.arpa; type=PTR
   2.30 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=185.110.23.0; name=9.20.110.185.in-addr.arpa; type=PTR
   2.36 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=0.23.110.185.in-addr.arpa; type=PTR
   2.40 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=0.23.110.185.in-addr.arpa; type=PTR
   2.41 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.253.249.53; name=0.23.110.185.in-addr.arpa; type=PTR
   2.57 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=185.110.23.0; name=0.23.110.185.in-addr.arpa; type=PTR
   2.63 INFO      ADDRESS02      NAMESERVERS_IP_WITH_REVERSE   
   2.63 DEBUG     ADDRESS02      TEST_CASE_END   testcase=address02
   2.63 DEBUG     UNSPECIFIED    MODULE_END   module=Address
```
```
$ zonemaster-cli 15.248.151.in-addr.arpa --raw --show_testcase --test address/address03 --level DEBUG
   0.00 DEBUG     UNSPECIFIED    START_TIME   string=2022-07-25 16:55:13 +0000; time_t=1658768113
   0.00 DEBUG     UNSPECIFIED    TEST_ARGS   args=Zonemaster::Engine::Zone=HASH(0x804fecb40); module=address; testcase=address03
   0.00 INFO      UNSPECIFIED    GLOBAL_VERSION   version=v4.5.1
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=Zonemaster::LDNS; version=2.2.2
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=IO::Socket::INET6; version=2.72
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=Moose; version=2.2201
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=Module::Find; version=0.15
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=File::ShareDir; version=1.118
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=File::Slurp; version=9999.27
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=Net::IP; version=1.26
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=List::MoreUtils; version=0.430
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=Clone; version=0.45
   0.00 DEBUG     UNSPECIFIED    DEPENDENCY_VERSION   name=Readonly; version=2.05
   0.00 DEBUG     UNSPECIFIED    MODULE_VERSION   module=Zonemaster::Engine::Test::Address; version=v1.0.8
   0.00 DEBUG     ADDRESS03      TEST_CASE_START   testcase=address03
   0.01 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=15.248.151.in-addr.arpa; type=SOA
   0.04 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=15.248.151.in-addr.arpa; type=SOA
   0.05 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=rirns.arin.net; type=A
   0.08 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:501:b1f9::30; name=rirns.arin.net; type=A
   0.09 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=204.61.216.50; name=rirns.arin.net; type=A
   0.10 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=rirns.arin.net; type=AAAA
   0.14 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:501:b1f9::30; name=rirns.arin.net; type=AAAA
   0.14 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=204.61.216.50; name=rirns.arin.net; type=AAAA
   0.16 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.253.249.53; name=15.248.151.in-addr.arpa; type=SOA
   0.31 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=ns1.nbit.it; type=A
   0.35 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:1010:7::53; name=ns1.nbit.it; type=A
   0.35 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=5.196.13.82; name=ns1.nbit.it; type=A
   0.38 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=ns1.nbit.it; type=AAAA
   0.42 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:1010:7::53; name=ns1.nbit.it; type=AAAA
   0.42 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=5.196.13.82; name=ns1.nbit.it; type=AAAA
   0.46 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=185.110.23.0; name=15.248.151.in-addr.arpa; type=SOA
   0.52 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.253.249.53; name=248.151.in-addr.arpa; type=SOA
   0.67 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=151.in-addr.arpa; type=SOA
   0.70 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=151.in-addr.arpa; type=SOA
   0.72 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.253.249.53; name=151.in-addr.arpa; type=SOA
   0.87 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=in-addr.arpa; type=SOA
   0.91 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=in-addr.arpa; type=SOA
   0.92 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=arpa; type=SOA
   0.95 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=198.41.0.4; name=arpa; type=NS
   0.96 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=a.ns.arpa; type=A
   0.99 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=a.ns.arpa; type=AAAA
   1.03 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=198.41.0.4; name=in-addr.arpa; type=NS
   1.03 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=a.in-addr-servers.arpa; type=A
   1.07 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=a.in-addr-servers.arpa; type=A
   1.07 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=a.in-addr-servers.arpa; type=AAAA
   1.10 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=a.in-addr-servers.arpa; type=AAAA
   1.11 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.180.182.53; name=in-addr.arpa; type=NS
   1.27 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.180.182.53; name=151.in-addr.arpa; type=NS
   1.42 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=ns3.afrinic.net; type=A
   1.46 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:501:b1f9::30; name=ns3.afrinic.net; type=A
   1.47 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2620:38:2000::53; name=ns3.afrinic.net; type=A
   1.62 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=ns3.afrinic.net; type=AAAA
   1.66 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:501:b1f9::30; name=ns3.afrinic.net; type=AAAA
   1.67 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2620:38:2000::53; name=ns3.afrinic.net; type=AAAA
   1.82 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:500:14:6100:ad::1; name=151.in-addr.arpa; type=NS
   1.84 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:500:14:6100:ad::1; name=15.248.151.in-addr.arpa; type=NS
   1.86 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=ns1.extnet.it; type=A
   1.89 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:1010:7::53; name=ns1.extnet.it; type=A
   1.90 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=5.196.13.82; name=ns1.extnet.it; type=A
   1.94 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=ns1.extnet.it; type=AAAA
   1.97 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:1010:7::53; name=ns1.extnet.it; type=AAAA
   1.97 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=5.196.13.82; name=ns1.extnet.it; type=AAAA
   2.01 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=185.110.20.9; name=15.248.151.in-addr.arpa; type=NS
   2.08 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=9.20.110.185.in-addr.arpa; type=PTR
   2.11 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=9.20.110.185.in-addr.arpa; type=PTR
   2.13 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.253.249.53; name=9.20.110.185.in-addr.arpa; type=PTR
   2.29 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=185.110.23.0; name=9.20.110.185.in-addr.arpa; type=PTR
   2.35 NOTICE    ADDRESS03      NAMESERVER_IP_PTR_MISMATCH   names=s1.extnet.it.; ns_ip=185.110.20.9; nsname=ns1.extnet.it
   2.35 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=0.23.110.185.in-addr.arpa; type=PTR
   2.38 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=0.23.110.185.in-addr.arpa; type=PTR
   2.40 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.253.249.53; name=0.23.110.185.in-addr.arpa; type=PTR
   2.56 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=185.110.23.0; name=0.23.110.185.in-addr.arpa; type=PTR
   2.65 NOTICE    ADDRESS03      NAMESERVER_IP_PTR_MISMATCH   names=uo1.nbit.it.; ns_ip=185.110.23.0; nsname=ns1.nbit.it
   2.65 DEBUG     ADDRESS03      TEST_CASE_END   testcase=address03
   2.65 DEBUG     UNSPECIFIED    MODULE_END   module=Address
```

**Comment by matsduf** [comment]
Something is strange with these lines from output from @anandb-ripencc. The query name is "23.110.185.in-addr.arpa", not the expected "0.23.110.185.in-addr.arpa". Compare with other PTR queries and the output from me.
```
   1.47 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=23.110.185.in-addr.arpa; type=PTR
   1.48 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=23.110.185.in-addr.arpa; type=PTR
   1.50 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.253.249.53; name=23.110.185.in-addr.arpa; type=PTR
   1.59 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=185.110.23.0; name=23.110.185.in-addr.arpa; type=PTR
```
```
   1.50 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:dc3::35; name=23.110.185.in-addr.arpa; type=PTR
   1.52 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=2001:67c:e0::1; name=23.110.185.in-addr.arpa; type=PTR
   1.53 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.253.249.53; name=23.110.185.in-addr.arpa; type=PTR
   1.63 DEBUG     ADDRESS03      EXTERNAL_QUERY   flags={"class":"IN"}; ip=185.110.23.0; name=23.110.185.in-addr.arpa; type=PTR
```

**Comment by matsduf** [comment]
@anandb-ripencc, I guess that you use a custom profile. Could you upload that or send that to me in an email? Also, the content of any local or global `cli.args`.

**Comment by matsduf** [comment]
The tests I reported above where run on FreeBSD. When I instead ran it on Docker, which is some Linux, I can replicate the error that you report. The owner name in the PTR query is wrong.

What OS do you run on?

```
$ docker run -t --rm zonemaster/cli 15.248.151.in-addr.arpa --raw --show_testcase --test address/address02 --level DEBUG --noipv6 
(...)
   4.15 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=202.12.27.33; name=23.110.185.in-addr.arpa; type=PTR
   4.22 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=193.0.9.1; name=23.110.185.in-addr.arpa; type=PTR
   4.31 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.253.249.53; name=23.110.185.in-addr.arpa; type=PTR
   4.50 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=185.110.23.0; name=23.110.185.in-addr.arpa; type=PTR
   4.59 WARNING   ADDRESS02      NAMESERVER_IP_WITHOUT_REVERSE   ns_ip=185.110.23.0; nsname=ns1.nbit.it
   4.60 DEBUG     ADDRESS02      TEST_CASE_END   testcase=address02
   4.60 DEBUG     UNSPECIFIED    MODULE_END   module=Address
```

**Comment by matsduf** [comment]
When testing on Debian there was no issue.

**Comment by matsduf** [comment]
I can confirm that with CentOS 7 Linux I also see the issue:
```
   2.33 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=202.12.27.33; name=9.20.110.185.in-addr.arpa; type=PTR
   2.36 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=193.0.9.1; name=9.20.110.185.in-addr.arpa; type=PTR
   2.38 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.253.249.53; name=9.20.110.185.in-addr.arpa; type=PTR
   2.55 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=185.110.23.0; name=9.20.110.185.in-addr.arpa; type=PTR
   2.61 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=202.12.27.33; name=23.110.185.in-addr.arpa; type=PTR
   2.65 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=193.0.9.1; name=23.110.185.in-addr.arpa; type=PTR
   2.67 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=199.253.249.53; name=23.110.185.in-addr.arpa; type=PTR
   2.83 DEBUG     ADDRESS02      EXTERNAL_QUERY   flags={"class":"IN"}; ip=185.110.23.0; name=23.110.185.in-addr.arpa; type=PTR
   2.90 WARNING   ADDRESS02      NAMESERVER_IP_WITHOUT_REVERSE   ns_ip=185.110.23.0; nsname=ns1.nbit.it
   2.90 DEBUG     ADDRESS02      TEST_CASE_END   testcase=address02
   2.90 DEBUG     UNSPECIFIED    MODULE_END   module=Address
```

Just as above, the first reverse name is correct, 9.20.110.185.in-addr.arpa. The second one is "23.110.185.in-addr.arpa" instead of expected "0.23.110.185.in-addr.arpa". Could it be that some code thinks that reverse of 185.110.23.0 should be the reverse of 185.110.23 since it ends with 0?

**Comment by anandb-ripencc** [comment]
Hi Mats. I am running on CentOS 7, and it certainly looks like something is using the wrong owner name for the PTR lookup of 185.110.23.0.

**Comment by matsduf** [comment]
The bug has been isolated and is reported in #1107.

**Comment by matsduf** [comment]
Resolved by #1108

**Comment by matsduf** [comment]
Resolved by #1119
